warehouse/forklift/legacy.py

```diff
@@ -384,47 +411,56 @@ class MetadataForm(forms.Form):
             ),
         ]
     )
-    comment = wtforms.StringField(validators=[wtforms.validators.Optional()])
+    comment = wtforms.StringField(
+        label="Comment",
```

Comment:
This field (and a number of others) don't correspond to actual core metadata fields as specified at https://packaging.python.org/specifications/core-metadata/.

I think it would be somewhat confusing if these fields failed to validate and returned an error message that looks like it's referring to a core metadata field (including linking to the spec) but that field doesn't actually exist.

I think we should probably drop the `label` attribute from any of these that are not in the spec, and update the error message logic to only raise the "new" error message (which links to the spec) if the field has a label.

Reply:
Mmm, so do you think a good idea would be to change the code to something like this?
``` python
# Check both if the field is present or it has a label to show the 'new' error
if field_name in form and form[field_name].label:
    raise _exc_with_message(
        HTTPBadRequest,
        "{value!r} is an invalid value for {field},".format(
            value=form[field_name].data,
            field=form[field_name].label.text) +
        "see https://packaging.python.org/specifications/core-metadata/",
    )
else:
   # Fall back to the 'old' error in a more friendly manner
    raise _exc_with_message(
        HTTPBadRequest,
        "Error: {}".format(form.errors[field_name][0])
    )
```

Reply:
The only issue with that is that it omits the field name that the original message had. I think there's three cases to consider here:

* the field name is in the form and has a label (it's a core metadata field) -> new error with link to spec
* the field name is in the form but doesn't have a label (it's one of these) -> original error
* the field name isn't in the form (it's `__all__`) -> new error without field name or link to spec